### PR TITLE
Reintroduce reading init_files for GEN_KW

### LINKED ISSUE #4055: GEN_KW INIT_FILES is used by an asset. We have to reintroduce it.

### PR DESCRIPTION
**Pull request by oyvindeide**
**Issue**
Resolves #4055

## Pre review checklist

- [x] Added appropriate release note label
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

Adding labels helps the maintainers when writing release notes. This is the [list of release note labels](https://github.com/equinor/ert/labels?q=release-notes).

### CODE REVIEW

File: tests/unit_tests/c_wrappers/integration/test_parameter_sample_types.py [hunk 1]
```diff
@@ -52,6 +53,18 @@ def load_from_forward_model(ert):
             [],
             does_not_raise(),
         ),
+        # (
+        #     "GEN_KW KW_NAME template.txt kw.txt prior.txt INIT_FILES:custom_param%d",
+        #     "MY_KEYWORD 1.31",
+        #     [("custom_param0", "MY_KEYWORD 1.31")],
```
Comment (oyvindeide) [inline tests/unit_tests/c_wrappers/integration/test_parameter_sample_types.py:59]: Not sure if this functionality is needed, currently checking

File: tests/unit_tests/c_wrappers/integration/test_parameter_sample_types.py [hunk 2]
```diff
@@ -218,16 +231,24 @@ def test_gen_kw_forward_init(tmpdir, load_forward_init):
...
+                ),
+            ):
+                create_runpath("config.ert")
+        else:
+            ert = create_runpath("config.ert")
+            assert Path("simulations/realization-0/iter-0/kw.txt").exists()
+            assert load_from_forward_model(ert) == 1
+            facade = LibresFacade(ert)
+            df = facade.load_all_gen_kw_data("default_0")
+            assert pd.isna(df["KW_NAME:MY_KEYWORD"][0])
```
Comment (oyvindeide) [inline tests/unit_tests/c_wrappers/integration/test_parameter_sample_types.py:251]: Not entirely clear to me why this should be `nan`

Comment (oyvindeide) [reply tests/unit_tests/c_wrappers/integration/test_parameter_sample_types.py]: It was reading it from a non-existing case...

File: tests/unit_tests/c_wrappers/integration/test_parameter_sample_types.py (hunk 1 above)
Comment (oyvindeide) [reply tests/unit_tests/c_wrappers/integration/test_parameter_sample_types.py]: It is needed

**Review by dafeda — APPROVED** [review]
LGTM 👍

**Pull request merged**